src/cloudant/index.py

```diff
@@ -262,3 +267,201 @@ def delete(self):
         """
         msg = 'Deleting the \"special\" index is not allowed.'
         raise CloudantException(msg)
+
+class SearchIndex(dict):
+    """
+    Encapsulates a SearchIndex as a dictionary based object, exposing the
+    search index function and analyzer as attributes. A SearchIndex object is
+    instantiated with a reference to a DesignDocument and is typically used as
+    part of the :class:`~cloudant.design_document.DesignDocument`
+    search index management API.
+
+    :param DesignDocument ddoc: DesignDocument instance used in part to
+        identify the search index.
+    :param str search_index_name: Name used in part to identify the index.
+    :param str search_index_func: Javascript search index function.
+        Optional only if executing a search query.
+    :param str analyzer: Optional analyzer of the index.  Defaults to standard.
+    """
+
+    def __init__(
+            self,
+            ddoc,
+            index_name,
+            search_func=None,
+            analyzer='standard'
+        ):
+        super(SearchIndex, self).__init__()
+        self.design_doc = ddoc
+        self.index_name = index_name
+        if search_func is not None:
+            self['index'] = codify(search_func)
+        self['analyzer'] = analyzer
+
+    @property
+    def analyzer(self):
+        """
+        Get the analyzer for this index.  Default value is a ``standard``
+        analyzer.  For more details on supported analyzers for a Cloudant
+        search index, see the analyzer
+        `documentation <https://docs.cloudant.com/search.html#analyzers>`_.
+
+        :param str analyzer: Analyzer for this index.
+
+        :returns: Search index analyzer
+        """
+        return self.get('analyzer')
+
+    @analyzer.setter
+    def analyzer(self, analyzer):
+        """
+        Set the analyzer for this index.
+        """
+        self['analyzer'] = analyzer
+
+    @property
+    def index(self):
+        """
+        Get the Javascript function for this index.
+
+        For example:
+
+        .. code-block:: python
+
+            # Set the SearchIndex index property
+            search.index = 'function (doc) {  index(\"default\", doc._id); }'
+
+        :param str index_func: Javascript search index function.
+
+        :returns: Codified search index function
+        """
+        return self.get('index')
+
+    @index.setter
+    def index(self, index_func):
```

Comment:
You probably want the `index_func` to be `search_func` to match what it is called in the constructor.